### PR TITLE
docs consistent with Dataset constructor taking variables

### PR DESCRIPTION
**Pull request by MaximilianR**
I think these should be `variables` rather than `data_vars` / `data`?

### CODE REVIEW

File: doc/data-structures.rst [hunk 1]
```diff
@@ -217,7 +217,7 @@ To make an :py:class:`~xray.Dataset` from scratch, supply dictionaries for any
 variables, coordinates and attributes you would like to insert into the
 dataset.
 
-For the ``vars`` and ``coords`` arguments, keys should be the name of the
+For the ``variables`` and ``coords`` arguments, keys should be the name of the
```
Comment (shoyer) [inline doc/data-structures.rst]: this change is correct

File: doc/data-structures.rst [hunk 2]
```diff
@@ -183,9 +183,9 @@ to access any variable in a dataset, datasets have four key properties:
 
 - ``dims``: a dictionary mapping from dimension names to the fixed length of
   each dimension (e.g., ``{'x': 6, 'y': 6, 'time': 8}``)
-- ``data_vars``: a dict-like container of DataArrays corresponding to variables
+- ``variables``: a dict-like container of DataArrays corresponding to variables
 - ``coords``: another dict-like container of DataArrays intended to label points
-  used in ``data_vars`` (e.g., 1-dimensional arrays of numbers, datetime
+  used in ``variables`` (e.g., 1-dimensional arrays of numbers, datetime
```
Comment (shoyer) [inline doc/data-structures.rst]: but these are not

**Comment by shoyer** [comment]
We should document `.variables` more clearly. It is indeed the fundamental property we use under the covers, but variables in `.variables` are stored as `xray.Variable` objects which is a low level API -- hence why we point people to `.data_vars` and `.coords` instead.

**Comment by maximilianr** [comment]
Ah OK, I'll make that change. 
I think it'd be clearer & more consistent if the args in the constructor & the attributes on the object were mirrored. Although looks like there's a lot going on there and is more complicated than it might seem.

**Comment by shoyer** [comment]
Hmm. Maybe it *would* be a good idea to rename `variables` to `data_vars` in the constructor. The argument name is really a relic of the old xray data model which didn't distinguish between data_vars and coords.

**Comment by maximilianr** [comment]
Happy to make that change as part of this - LMK. 
I'd keep the old one around as a deprecated kwarg

**Comment by maximilianr** [comment]
@shoyer made some changes to the `DataArray` construction docs - let me know if those are in the right direction. I imagine there's still some way to go there.

**Comment by shoyer** [comment]
Yes, this is great -- thank you!

I do think it would be a good idea to rename that first `Dataset` parameter to `data_vars`. That would certainly be more consistent.

**Comment by maximilianr** [comment]
@shoyer How's that? I think there some more stuff to do on the docs, but that's a stable upgrade for now? Very open to feedback, docs are definitely not my strong point.

**Comment by maximilianr** [comment]
@shoyer updated

**Comment by MaximilianR** [comment]
@shoyer let me know any feedback / changes

File: xray/core/dataset.py [hunk 3]
```diff
@@ -312,12 +312,19 @@ def __init__(self, variables=None, coords=None, attrs=None,
...
         self._file_obj = None
-        if variables is None:
-            variables = {}
+        if kwargs:
+            if 'variables' in kwargs:
+                data_vars = kwargs.pop('variables')
+                import warnings
+                warnings.warn('Variables kwarg is deprecated - use data_vars')
+            if kwargs:
+                raise AttributeError('{0} are not valid kwargs'.format(kwargs.keys()))
```
Comment (shoyer) [inline xray/core/dataset.py]: This should be TypeError

File: xray/core/dataset.py [hunk 4]
```diff
@@ -312,12 +312,19 @@ def __init__(self, variables=None, coords=None, attrs=None,
         self._dims = {}
         self._attrs = None
         self._file_obj = None
-        if variables is None:
-            variables = {}
+        if kwargs:
+            if 'variables' in kwargs:
+                data_vars = kwargs.pop('variables')
+                import warnings
+                warnings.warn('Variables kwarg is deprecated - use data_vars')
```
Comment (shoyer) [inline xray/core/dataset.py]: Use stacklevel=2 (I think) to raise this warning at the right level

File: doc/data-structures.rst [hunk 5]
```diff
@@ -42,50 +42,59 @@ to interpret them, and propagates them only in unambiguous cases (see FAQ,
...
     foo
 
-All of these arguments (except for ``data``) are optional, and will be filled
+Only ``data_vars`` is required; all of other arguments will be filled
 in with default values:
 
 .. ipython:: python
 
-    xray.DataArray(data)
+    xray.DataArray(data_vars)
```
Comment (shoyer) [inline doc/data-structures.rst]: The DataArray argument is still `data`, not `data_vars`

File: xray/core/dataset.py [hunk 6]
```diff
@@ -312,12 +312,19 @@ def __init__(self, variables=None, coords=None, attrs=None,
         self._dims = {}
         self._attrs = None
         self._file_obj = None
-        if variables is None:
-            variables = {}
+        if kwargs:
+            if 'variables' in kwargs:
+                data_vars = kwargs.pop('variables')
+                import warnings
```
Comment (shoyer) [inline xray/core/dataset.py]: Do this import at the top of this file (same for the import in the tests)

**Comment by shoyer** [comment]
Thanks for the reminder -- just a few minor corrections

**Comment by MaximilianR** [comment]
@shoyer updated & green

File: doc/data-structures.rst [hunk 7]
```diff
@@ -42,19 +42,25 @@ to interpret them, and propagates them only in unambiguous cases (see FAQ,
 Creating a DataArray
 ~~~~~~~~~~~~~~~~~~~~
 
-The :py:class:`~xray.DataArray` constructor takes a multi-dimensional array of
-values (e.g., a numpy ndarray), a list or dictionary of coordinates label and
-a list of dimension names:
+The :py:class:`~xray.DataArray` constructor takes:
+
+ - ``data_vars``: a multi-dimensional array of values (e.g., a numpy ndarray,
```
Comment (shoyer) [inline doc/data-structures.rst]: should still be `data`

File: doc/data-structures.rst [hunk 8]
```diff
@@ -42,19 +42,25 @@ to interpret them, and propagates them only in unambiguous cases (see FAQ,
...
 
     data = np.random.rand(4, 3)
     locs = ['IA', 'IL', 'IN']
     times = pd.date_range('2000-01-01', periods=4)
-    foo = xray.DataArray(data, coords=[times, locs], dims=['time', 'space'])
+    foo = xray.DataArray(data_vars, coords=[times, locs], dims=['time', 'space'])
     foo
 
-All of these arguments (except for ``data``) are optional, and will be filled
+Only ``data_vars`` is required; all of other arguments will be filled
```
Comment (shoyer) [inline doc/data-structures.rst]: still `data`

File: doc/data-structures.rst [hunk 9]
```diff
@@ -65,27 +71,30 @@ As you can see, dimensions and coordinate arrays corresponding to each
...
+- A dictionary of ``{coord_name: coord}`` where the values are scaler values,
+ 1D arrays or tuples (tuples in the same form as above). This form lets you supply other
+ coordinates than those corresponding to dimensions (more on these later).
+
+As a list of tuples:
 
 .. ipython:: python
 
-    xray.DataArray(data, coords=[('time', times), ('space', locs)])
+    xray.DataArray(data_vars, coords=[('time', times), ('space', locs)])
```
Comment (shoyer) [inline doc/data-structures.rst]: still `data`

File: doc/data-structures.rst [hunk 10]
```diff
@@ -65,27 +71,30 @@ As you can see, dimensions and coordinate arrays corresponding to each
...
-Yet another option is to supply ``coords`` in the form of a dictionary where
-the values are scaler values, 1D arrays or tuples (in the same form as the
-`dataarray constructor`_). This form lets you supply other coordinates than
-those corresponding to dimensions (more on these later):
+As a dictionary:
 
 .. ipython:: python
 
-    xray.DataArray(data, coords={'time': times, 'space': locs, 'const': 42,
+    xray.DataArray(data_vars, coords={'time': times, 'space': locs, 'const': 42,
```
Comment (shoyer) [inline doc/data-structures.rst]: still `data`

File: doc/data-structures.rst (hunk 1 above)
Comment (shoyer) [inline doc/data-structures.rst]: should be `data_vars` now

File: doc/data-structures.rst [hunk 11]
```diff
@@ -217,12 +226,12 @@ To make an :py:class:`~xray.Dataset` from scratch, supply dictionaries for any
...
 dataset.
 
-For the ``vars`` and ``coords`` arguments, keys should be the name of the
+For the ``variables`` and ``coords`` arguments, keys should be the name of the
 variable and values should be scalars, 1d arrays or tuples of the form
 ``(dims, data[, attrs])`` sufficient to label each array:
 
 - ``dims`` should be a sequence of strings.
-- ``data`` should be a numpy.ndarray (or array-like object) that has a
+- ``data_vars`` should be a numpy.ndarray (or array-like object) that has a
```
Comment (shoyer) [inline doc/data-structures.rst]: `data`

**Comment by shoyer** [comment]
Sorry still a few places where the terminology was mixed up. We aren't renaming the `DataArray` argument here :)

**Comment by MaximilianR** [comment]
Apologies! Will fix

**Comment by MaximilianR** [comment]
@shoyer I think this is ready now

**Comment by shoyer** [comment]
thanks!

**Pull request merged**